src/inmanta/agent/handler.py

```diff
@@ -932,6 +933,14 @@ def execute(self, ctx: HandlerContext, resource: resources.Resource, dry_run: Op
                 )
 
 
+# Keep the CRUDHandler class for backwards compatibility. Making the CRUDHandler class generic can cause MRO issues
+# if a subclass of CRUDHandler is generic as well. The code below makes a copy of the CRUDHandlerGeneric class
+# and removes Generic as a base class.
+CRUDHandler = types.new_class(
```

Comment:
Iirc my testing revealed that it isn't a problem per se that it inherits from `Generic`, more that it inherited from a non-parametrized generic. I think we should be fine with
```python
class CRUDHandler(CRUDHandlerGeneric[resources.PurgeableResource]): pass

Reply:
No. The issue is the method resolution order (MRO) that cannot be determined. The inheritance structure looks like this:

```mermaid
classDiagram
    class Generic
    class CRUDHadlerGeneric
    class CRUDHandler
    class BaseHandler
    class CFSOrderResourceHandler

    Generic <|-- CRUDHadlerGeneric
    CRUDHadlerGeneric <|-- CRUDHandler
    CRUDHandler <|-- BaseHandler
    BaseHandler <|-- CFSOrderResourceHandler
    Generic <|-- CFSOrderResourceHandler
```

The `CFSOrderResourceHandler` class is defined as:

```
class CFSOrderResourceHandler(Generic[Order], BaseHandler):
```

which makes it impossible to find a consistent ordering. If the signature would be changed to:

```
class CFSOrderResourceHandler(BaseHandler, Generic[Order])
```

it works. But that is not going to help us here.

I tested with your proposal and the MRO issue is still there.